servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -6726,6 +6726,46 @@ public void putTenancy(ResourceContext ctx, String tenantDomain, String provider
         if (authzServiceTokenOperation) {
             setupTenantAdminPolicyInProvider(ctx, provSvcDomain, provSvcName, tenantDomain,
                     false, auditRef, caller);
+
+            // If listed as service provider, register dependency between the provider service and the tenant domain (unless dependency exist
+            // following creation of a previous resource group)
+
+            tenancyRegisterDomainDependency(ctx, tenantDomain, provider, auditRef, caller);
+        }
+    }
+
+    private void tenancyRegisterDomainDependency(ResourceContext ctx, String tenantDomain, String provider, String auditRef, String caller) {
+        try {
+            if (serviceProviderManager.isServiceProvider(provider)) {
+
+                DomainList domainList = dbService.listDomainDependencies(provider);
+                if (!domainList.getNames().contains(tenantDomain)) {
+                    dbService.putDomainDependency(ctx, tenantDomain, provider, auditRef, caller);
+                }
+            }
+        } catch (ResourceException ex) {
```

Comment:
do we still need this try/catch block? I don't see which api is returning the unauthorized exception.